src/NServiceBus.Core/Persistence/Learning/SagaPersister/SagaStorageFile.cs

```diff
@@ -88,20 +80,21 @@ public Task MarkAsCompleted()
             return TaskEx.CompletedTask;
         }
 
-        public object Read()
+        public async Task<object> Read()
         {
-            return serializer.Deserialize(jsonReader, manifest.SagaEntityType);
+            var json = await streamReader.ReadToEndAsync().ConfigureAwait(false);
+            return SimpleJson.DeserializeObject(json, manifest.SagaEntityType, SerializationStrategy);
         }
 
 
         SagaManifest manifest;
         FileStream fileStream;
         bool isCompleted;
-        JsonTextWriter jsonWriter;
-        JsonTextReader jsonReader;
+        StreamWriter streamWriter;
+        StreamReader streamReader;
 
         const int DefaultBufferSize = 4096;
-        static Newtonsoft.Json.JsonSerializer serializer = new Newtonsoft.Json.JsonSerializer();
         static Task<SagaStorageFile> noSagaFoundResult = Task.FromResult<SagaStorageFile>(null);
+        static readonly IJsonSerializerStrategy SerializationStrategy = new EnumAwareStrategy();
```

Comment:
It would also be possible to set `SimpleJson.Current...` property. I decided to make it opt-in per usage. Another option would be to introduce a static property on `EnumAwareStrategy`. Although I think we don't really need to save memory here in such an aggressive way.